### PR TITLE
planner: add warnings about joins when tidb_enforce_mpp is enabled.

### PR DESCRIPTION
**Pull request by LittleFall**
### What problem does this PR solve?

Issue Number: close https://github.com/pingcap/tics/issues/1905 

Problem Summary: This is the follow-up of https://github.com/pingcap/tidb/pull/25302, add warnings when tidb_enforce_mpp is enabled but mpp is blocked.

### What is changed and how it works?

What's Changed: adding warnings for join.

How it Works:

### Check List 

Tests 

- [x] Unit test
- [x] Integration test

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note 

- No release note

### CODE REVIEW

**Review by hanfei1991 — APPROVED** [review]

File: planner/core/exhaust_physical_plans.go [hunk 1]
```diff
@@ -1697,7 +1696,8 @@ func (p *LogicalJoin) exhaustPhysicalPlans(prop *property.PhysicalProperty) ([]P
 		}
 	})
 
-	if prop.IsFlashProp() && ((p.preferJoinType&preferBCJoin) == 0 && p.preferJoinType > 0) {
+	if prop.IsFlashProp() && (p.preferJoinType&preferBCJoin) == 0 && p.preferJoinType > 0 {
+		p.SCtx().GetSessionVars().RaiseWarningWhenMPPEnforced("MPP mode may be blocked because you have used hint to specify a join algorithm which is not supported by mpp now.")
```
Comment (windtalker) [inline planner/core/exhaust_physical_plans.go:1700]: This is a little confusing, user use `broadcast_join` hint which is only supported in mpp mode.

File: planner/core/exhaust_physical_plans.go [hunk 2]
```diff
@@ -1786,15 +1786,19 @@ func (p *LogicalJoin) tryToGetMppHashJoin(prop *property.PhysicalProperty, useBC
...
 	if len(p.EqualConditions) == 0 {
-		if p.ctx.GetSessionVars().AllowCartesianBCJ == 0 || !useBCJ {
+		if !useBCJ || p.ctx.GetSessionVars().AllowCartesianBCJ == 0 {
+			// these warnings assume users won't change variables `tidb_opt_broadcast_join`.
+			p.SCtx().GetSessionVars().RaiseWarningWhenMPPEnforced("MPP mode may be blocked because `Cartesian Product` is only supported by broadcast join, check value and documents of variables `tidb_opt_broadcast_cartesian_join`, `tidb_broadcast_join_threshold_size` and `tidb_broadcast_join_threshold_count`.")
 			return nil
 		}
 	}
 	if (len(p.LeftConditions) != 0 && p.JoinType != LeftOuterJoin) || (len(p.RightConditions) != 0 && p.JoinType != RightOuterJoin) {
+		p.SCtx().GetSessionVars().RaiseWarningWhenMPPEnforced("MPP mode may be blocked because there is a join that is not `left join` but has left conditions, which is not supported by mpp now, see github.com/pingcap/tidb/issues/26090 for more information.")
```
Comment (windtalker) [inline planner/core/exhaust_physical_plans.go:1801]: right condition with non-right join is also disabled.

Comment (LittleFall) [reply planner/core/exhaust_physical_plans.go:1801]: ok, and is there any example for `right condition with non-right join`?

File: planner/core/exhaust_physical_plans.go (hunk 1 above)
Comment (LittleFall) [reply planner/core/exhaust_physical_plans.go:1700]: `broadcast_join` won't block this

**Review by windtalker — APPROVED** [review]
LGTM

**Pull request merged**